src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+import { Person } from '../types/Person';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
+import './Autocomplete.scss';
+
+type Props = {
+  onSelected: (person: Person) => void;
+  people: Person[];
+  cleareSelectedPerson: () => void;
+  delay?: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  onSelected,
+  people,
+  cleareSelectedPerson,
+  delay = 300,
+}) => {
+  const [inputChange, setInputChange] = useState('');
+  const [currentPeople, setCurrentPeople] = useState(people);
+  const [selectPerson, setSelectPerson] = useState<Person | null>(null);
+
+  const personSelect = (person: Person) => {
+    setSelectPerson(person);
+    setInputChange(person.name);
+    setCurrentPeople([]);
+    onSelected(person);
+  };
+
+  const debouncedFilterPeople = debounce((inputValue: string) => {
+    const filteredPeople = people.filter(person =>
+      person.name
+        .toLowerCase()
+        .trim()
+        .includes(inputValue.toLowerCase().trim()),
+    );
+
+    setCurrentPeople(filteredPeople);
+  }, delay);
+
+  const handleSelect = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+
+    setInputChange(value);
+
+    if (selectPerson && value !== selectPerson.name) {
+      setSelectPerson(null);
+      cleareSelectedPerson();
+    }
+
+    debouncedFilterPeople(value);
+  };
+
+  return (
+    <div className="dropdown is-active">
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={inputChange}
+          onChange={handleSelect}
+          data-cy="search-input"
+        />
+      </div>
+      <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
+        <div
+          className={classNames('dropdown-content', {
+            hidden: selectPerson !== null,
+          })}
+        >
+          {currentPeople.length > 0 || selectPerson !== null ? (
```

Comment:
it should be visible only if the input is focused:

![image](https://github.com/user-attachments/assets/20c3992e-439e-4e6d-a7bd-f146240a2688)